extensions/notebook/src/book/bookTreeItem.ts

```diff
@@ -4,20 +4,92 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as vscode from 'vscode';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
 
 export class BookTreeItem extends vscode.TreeItem {
+	public sections: any[];
+	public uri: string;
+	public previousUri: string;
+	public nextUri: string;
+	public command: vscode.Command;
 
 	constructor(
-		public readonly title: string,
-		public readonly root: string,
-		public readonly tableOfContents: any[],
-		public readonly collapsibleState: vscode.TreeItemCollapsibleState,
-		public uri?: string,
-		public command?: vscode.Command
+		public title: string,
+		public root: string,
+		public tableOfContents: any[],
+		public page: any,
```

Comment:
It'd be great if you can define the expected interface for these JSON objects. That's what we do for the Jupyter notebook format JSON objects, and it makes it less likely that you'll have a typo of incorRect caSing cause a runtime error. Is it possible to add some interface definitions here so you can "safely" reference all the fields? Not required, but really helps us understand things / avoid future errors